src/service/resources-impl.js

```diff
@@ -298,12 +296,25 @@ export class ResourcesDef extends MutatorsAndOwnersDef {
    * @param {number} newLayoutPriority
    */
   updateLayoutPriority(element, newLayoutPriority) {}
+
+  /**
+   * Schedules layout or preload for the specified resource.
+   * @param {!Resource} resource
+   * @param {boolean} layout
+   * @param {number=} opt_parentPriority
+   * @param {boolean=} opt_forceOutsideViewport
+   */
+  scheduleLayoutOrPreload(
```

Comment:
`measureAndScheduleIfAllowed_` might be a better choice for public API than this method (less function params, and more code pushes to resources)
also consider rename it to `measureAndTrySchedule()`

Reply:
There are extensive dependencies on scheduleLayoutOrPreload on resources-impl.js

Ideas?